### ISSUE TITLE
Speech separation eval speed is soo slow

### ISSUE THREAD

**Issue by srdfjy** [speechbrain/speechbrain#963]
How to use gpu to improve the evaluation speed, although the use of torch-mir-eval 0.4 version has improved, but it has a torch conflict with speechbrain.

**Comment by srdfjy** [comment]
There is only one round in total, and the log training, verification, and test steps are completed. What is the red box? It also runs very slowly on the cpu.

**Comment by srdfjy** [comment]
![image](https://user-images.githubusercontent.com/66103286/131460605-522979b3-a3f1-4bae-b6f4-0d7e196e73a3.png)

How to improve the running speed of the box, the step of the red box is in this red, does anyone know? Thanks!

**Comment by srdfjy** [comment]
It should be your assessment that this step is particularly slow, running on the cpu, I was training is particularly fast on the gpu.

**Comment by srdfjy** [comment]
How to use gpu to improve the evaluation speed, although the use of torch-mir-eval 0.4 version has improved, but it has a torch conflict with speechbrain.

**Comment by mravanelli** [comment]
@ycemsubakan and @popcornell any insight on that?

**Comment by ycemsubakan** [comment]
Do you use pretrained models? Do you take the model to the gpu?

**Comment by srdfjy** [comment]
hi @ycemsubakan I run the default speech separation recipe，use sepformer model.

**Comment by ycemsubakan** [comment]
How long are the recordings you are trying to work with?

**Comment by srdfjy** [comment]
50 test samples, 60 core cpu, half completed in 1 hour.

2021/09/08 17:31:46   0%|          | 0/50 [00:00<?, ?it/s]
2021/09/08 17:34:17   2%|▏         | 1/50 [02:30<2:02:53, 150.47s/it]
2021/09/08 17:36:47   4%|▍         | 2/50 [05:01<2:00:31, 150.66s/it]
2021/09/08 17:40:18   6%|▌         | 3/50 [07:32<1:58:05, 150.76s/it]
2021/09/08 17:44:08   8%|▊         | 4/50 [11:02<2:13:40, 174.36s/it]
2021/09/08 17:46:46  10%|█         | 5/50 [14:53<2:25:56, 194.60s/it]
2021/09/08 17:50:39  12%|█▏        | 6/50 [17:30<2:13:23, 181.89s/it]
2021/09/08 17:53:52  14%|█▍        | 7/50 [21:23<2:22:22, 198.67s/it]
2021/09/08 17:56:20  16%|█▌        | 8/50 [24:36<2:17:43, 196.76s/it]
2021/09/08 17:58:46  18%|█▊        | 9/50 [27:04<2:04:06, 181.61s/it]
2021/09/08 18:01:09  20%|██        | 10/50 [29:30<1:53:43, 170.58s/it]
2021/09/08 18:03:35  22%|██▏       | 11/50 [31:53<1:45:24, 162.17s/it]
2021/09/08 18:05:57  24%|██▍       | 12/50 [34:19<1:39:40, 157.38s/it]
2021/09/08 18:08:24  26%|██▌       | 13/50 [36:41<1:34:06, 152.62s/it]
2021/09/08 18:10:53  28%|██▊       | 14/50 [39:08<1:30:36, 151.01s/it]
2021/09/08 18:13:21  30%|███       | 15/50 [41:37<1:27:41, 150.32s/it]
2021/09/08 18:15:44  32%|███▏      | 16/50 [44:05<1:24:47, 149.63s/it]
2021/09/08 18:18:14  34%|███▍      | 17/50 [46:28<1:21:12, 147.64s/it]
2021/09/08 18:20:37  36%|███▌      | 18/50 [48:58<1:19:02, 148.22s/it]
2021/09/08 18:23:02  38%|███▊      | 19/50 [51:22<1:15:55, 146.94s/it]
2021/09/08 18:25:29  40%|████      | 20/50 [53:46<1:13:05, 146.19s/it]
2021/09/08 18:28:01  42%|████▏     | 21/50 [56:14<1:10:50, 146.58s/it]
2021/09/08 18:30:28  44%|████▍     | 22/50 [58:45<1:09:06, 148.08s/it]
2021/09/08 18:32:56  46%|████▌     | 23/50 [1:01:12<1:06:25, 147.61s/it]
2021/09/08 18:35:23  48%|████▊     | 24/50 [1:03:40<1:04:04, 147.86s/it]

**Comment by ycemsubakan** [comment]
what are the lengths of each of these recordings?

**Comment by gkucsko** [comment]
seeing the same for spectral_mask recipe. looks almost like it's run on a single core rather than gpu. maybe some metrics are expensive and not multi-cored? (like pesq or stoi?)

**Comment by mravanelli** [comment]
Right, the pesq and stoi runs on cpu. As far as I remember, I saw that
issue in the speech enhancement part as well. My solution was to set the
metric computation to 1 core (like nj=0). This way it is much faster, but I
didn't have the chance to investigate too much on that.

**Comment by srdfjy** [comment]
> what are the lengths of each of these recordings?

2-5 milliseconds per file

**Comment by srdfjy** [comment]
I found that if the speakers in spk1 and spk2 do not overlap (that is, the speech in spk1 will not appear in spk2, and vice versa), the evaluation will be much faster. Does the sepformer model have any requirements for data set generation?

**Comment by ycemsubakan** [comment]
No we do not have any special requirements like this. I am not sure to understand what is going on here..

**Comment by srdfjy** [comment]
thanks @ycemsubakan @mravanelli
